File: apps/src/main/kotlin/io/dustin/apps/common/config/RigoWeb3Configuration.java
//package io.dustin.apps.common.config;
//
//import io.dustin.apps.common.properties.RigoProperties;
//import io.rigo.sdk.common.client.RigoWeb3;
//import io.rigo.sdk.common.client.factory.RigoApiClientFactory;
//import lombok.RequiredArgsConstructor;
//import org.springframework.context.annotation.Bean;
//import org.springframework.context.annotation.Configuration;
//
//@Configuration
//@RequiredArgsConstructor
//public class RigoWeb3Configuration {
//
//    private final RigoProperties prop;
//
//    @Bean
//    public RigoWeb3 rigoClient() {
//        RigoApiClientFactory factory = RigoApiClientFactory.newInstance(prop.getNodeUrl());
//        return factory.rigoWeb3();
//    }
//}

File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/common/config/RigoWeb3Configuration.kt
package io.dustin.apps.common.config

import io.dustin.apps.common.properties.RigoProperties

@Configuration
@RequiredArgsConstructor
class RigoWeb3Configuration {
    private val prop: RigoProperties? = null
    @Bean
    fun rigoClient(): RigoWeb3? {
        val factory: RigoApiClientFactory = RigoApiClientFactory.newInstance(prop.getNodeUrl())
        return factory.rigoWeb3()
    }
}
